### ISSUE TITLE
ImageLoader.asMatrix() or asRowVector() returns wrong values since 1.0.0-beta4

### ISSUE THREAD

**Issue by vincenzocaselli** [deeplearning4j/deeplearning4j#7809]
Using version **1.0.0-beta4** 
**ImageLoader.asMatrix()** or **asRowVector()** 
return wrong values. 
For instance using 
ImageLoader.asRowVector() on MNIST file 
mnist_png/training/3/11111.png
we get 
[[ -1.6777e7, -1.6777e7, ...
instead of
[[         0,         0,         0,   ...
(1.0.0-beta3)

Does anyone else confirms this or is it just me?
Thank you very much

**Comment by saudet** [comment]
Could you post the exact code you're using?

**Comment by Charele** [comment]
val file = "C:\\mnist_png\\training\\0\\359.png"  
new ImageLoader().asMatrix(new File(file)).show

output:
o.n.l.a.o.e.DefaultOpExecutioner - Cores: [8]; Memory: [3.5GB];
o.n.l.a.o.e.DefaultOpExecutioner - Blas vendor: [CUBLAS]
o.n.l.j.o.e.CudaExecutioner - Device Name: [GeForce GTX 1080 Ti]; CC: [6.1]; Total/free memory: [11811160064]
[[ -1.6777e7, -1.6777e7, -1.6777e7, -1.6777e7, -1.6777e7, -1.6777e7, -1.6777e7,

**Comment by saudet** [comment]
NativeImageLoader works fine. It seems to be a problem with ImageLoader and its call to NDArrayUtil.toNDArray(). @raver119 Any known problems with that?

**Comment by raver119** [comment]
Mmmm... This thing looks outdated, and doesn't support new datatypes and multiple default dtypes. Methods `toNDArray(...)` can be deprecated, since we have `Nd4j.createFromArray(...)` signatures now, covering all possible primitive types

**Comment by raver119** [comment]
Mmm, wasn't `ImageLoader` deprecated? AFAIK these days we use NativeImageLoader everywhere.

**Comment by saudet** [comment]
Yes, we're not adding new functionality, but some people still prefer to use ImageLoader to get less native dependencies, so we shouldn't break it.

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
